enter.c

```diff
@@ -230,31 +230,55 @@ int enter(struct entry_settings *opts)
 
 	pid_t rootpid = getpid();
 
-	int cgroupfd;
-
 	/*
 	 * If there is a cgroup specified we need to create bst.<pid> subcgroup to join
 	 * Otherwise if no cgroup is specified but limits are applied bst will error out
 	 */
-	if (opts->cgroup_path != NULL) {
+	if (opts->nactiveclimits != 0) {
+
+		// If cgroup path is unspecified then attempt to derive from /proc/self/cgroup
+		if (opts->cgroup_path == NULL) {
+			FILE *selfcgroupfd = fopen("/proc/self/cgroup", "r");
+			if (selfcgroupfd == NULL) {
+				err(1, "unable to derive current cgroup hierarchy from /proc/self/cgroup");
+			}
+
+			// Move past "0:/" cgroup root in /proc/self/cgroup to get path
+			if (fseek(selfcgroupfd, 3, SEEK_SET) != 0) {
```

Comment:
This file may have multiple lines when cgroupsv1 are involved (for instance, one cgroupv1 controller is mounted while v2 is also mounted). It's better to parse this line by line and search for the "0::" prefix.

It's also possible that this is a full cgroups v1 system and there is no path. This is not a cause for error, but we should simply warn and move on as if --cgroup was not specified and was mandatory.

Reply:
OK. Now searches each line for prefix.

Reply:
If there is no path to derive (on a cgroups v1 system) but there are --limits specified this should be a error right?

Reply:
Probably. It'll fail anyway to open the files in question since they're going to end up missing from the v2 hierarchy, so I'd be inclined to not explicitly fix it.